clients/admin-ui/src/features/openid-authentication/OpenIDProviderForm.tsx

```diff
@@ -0,0 +1,217 @@
+import { getErrorMessage, isErrorResult } from "~/features/common/helpers";
+import { errorToastParams, successToastParams } from "~/features/common/toast";
+import {
+  useCreateOpenIDProviderMutation,
+  useDeleteOpenIDProviderMutation,
+  useUpdateOpenIDProviderMutation,
+} from "~/features/openid-authentication/openprovider.slice";
+import { OpenIDProvider } from "~/types/api";
+
+interface OpenIDProviderFormProps {
+  openIDProvider?: OpenIDProvider;
+  onSuccess?: (openIDProvider: OpenIDProvider) => void;
+}
+
+export interface OpenIDProviderFormValues extends OpenIDProvider {}
+
+export const defaultInitialValues: OpenIDProviderFormValues = {
+  id: "",
+  provider: "",
+  client_id: "",
+  client_secret: "",
+};
+
+export const transformOrganizationToFormValues = (
+  openIDProvider: OpenIDProvider
+): OpenIDProviderFormValues => ({ ...openIDProvider });
+
+export const transformFormValuesToOpenIDProvider = (
+  formValues: OpenIDProviderFormValues
+): OpenIDProvider => ({
+  id: formValues.id,
+  provider: formValues.provider,
+  client_id: formValues.client_id,
+  client_secret: formValues.client_secret,
+});
+
+const OpenIDProviderFormValidationSchema = Yup.object().shape({
+  provider: Yup.string().required().label("Provider"),
+  client_id: Yup.string().required().label("Client ID"),
+  client_secret: Yup.string().required().label("Client Secret"),
+});
+
+export const OpenIDProviderForm = ({
+  openIDProvider,
+  onSuccess,
+}: OpenIDProviderFormProps) => {
+  const [createOpenIDProviderMutationTrigger] =
+    useCreateOpenIDProviderMutation();
+  const [updateOpenIDProviderMutation, updateOpenIDProviderMutationResult] =
+    useUpdateOpenIDProviderMutation();
+  const [deleteOpenIDProviderMutation] = useDeleteOpenIDProviderMutation();
+
+  const initialValues = useMemo(
+    () =>
+      openIDProvider
+        ? transformOrganizationToFormValues(openIDProvider)
+        : defaultInitialValues,
+    [openIDProvider]
+  );
+
+  const {
+    isOpen: deleteIsOpen,
+    onOpen: onDeleteOpen,
+    onClose: onDeleteClose,
+  } = useDisclosure();
+
+  const toast = useToast();
+
+  const handleSubmit = async (
+    values: OpenIDProviderFormValues,
+    formikHelpers: FormikHelpers<OpenIDProviderFormValues>
+  ) => {
+    const openIDProviderBody = transformFormValuesToOpenIDProvider(values);
+
+    const handleResult = (
+      result: { data: {} } | { error: FetchBaseQueryError | SerializedError }
+    ) => {
+      if (isErrorResult(result)) {
+        const errorMsg = getErrorMessage(
+          result.error,
+          "An unexpected error occurred while editing the OpenID Provider. Please try again."
+        );
+        toast(errorToastParams(errorMsg));
+      } else {
+        toast(successToastParams("OpenID Provider configuration saved."));
+        formikHelpers.resetForm({});
+        if (onSuccess) {
+          onSuccess(openIDProviderBody);
+        }
+      }
+    };
+    if (initialValues.id) {
```

Comment:
Can we consider creating a new variable called isEditing so that the condition is clearer?